### PR TITLE
Update docker

### PR DESCRIPTION
**Pull request by tianon**
Changes:

- https://github.com/docker-library/docker/commit/a4e8aab: Update 25-rc to buildx 0.12.1
- https://github.com/docker-library/docker/commit/e941e9d: Update 24 to buildx 0.12.1
- https://github.com/docker-library/docker/commit/72a378e: Update 25-rc to compose 2.24.0
- https://github.com/docker-library/docker/commit/ef1774b: Update 24 to compose 2.24.0
- https://github.com/docker-library/docker/commit/76e5e78: Merge pull request https://github.com/docker-library/docker/pull/472 from infosiftr/github-atom-json
- https://github.com/docker-library/docker/commit/e8fb8d7: Switch to consuming JSON from GitHub for "latest commit"

### CODE REVIEW

**Comment by tianon** [comment]
Oh, I see @thaJeztah is working on a v25.0.0-rc.2 literally as I open this; maybe we should wait :smile:

**Comment by thaJeztah** [comment]
It'll be tagged once it's on production (Irolling out to staging takes some hours, then again to prod), so 🤷

**Comment by thaJeztah** [comment]
And I did just notice that the buildx repo didn't update Go versions for some time, so build v0.12.1 packages (if those come from that repo)  are probably still built with go1.21.3 😞 (https://github.com/docker/buildx/pull/2190 / https://github.com/docker/buildx/pull/2191)

**Review by yosifkit — APPROVED** [review]

File: library/docker [hunk 1]
```diff
@@ -5,45 +5,45 @@ Maintainers: Tianon Gravi <tianon@dockerproject.org> (@tianon),
...
 
-Tags: 25.0.0-rc.1-git, 25-rc-git, rc-git
+Tags: 25.0.0-rc.2-git, 25-rc-git, rc-git
 Architectures: amd64, arm32v6, arm32v7, arm64v8
 GitCommit: 2e213030c57a2134a77bf17b0710dff1a184a7c1
 Directory: 25-rc/git
 
-Tags: 25.0.0-rc.1-windowsservercore-ltsc2022, 25-rc-windowsservercore-ltsc2022, rc-windowsservercore-ltsc2022
-SharedTags: 25.0.0-rc.1-windowsservercore, 25-rc-windowsservercore, rc-windowsservercore
+Tags: 25.0.0-rc.2-windowsservercore-ltsc2022, 25-rc-windowsservercore-ltsc2022, rc-windowsservercore-ltsc2022
```
Comment (thaJeztah) [inline library/docker:28]: ❤️ thanks!

**Pull request merged**